client/app/visualizations/Visualization.jsx

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable react/no-multi-comp */
+
+import { find } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import { Filters, FiltersType } from '@/components/Filters';
+
+export let Visualization = null; // eslint-disable-line import/no-mutable-exports
+
+const visualizations = {};
+
+// for `registerVisualization`
+export const VisualizationConfig = PropTypes.shape({
+  type: PropTypes.string.isRequired,
+  name: PropTypes.string.isRequired,
+  getOptions: PropTypes.func.isRequired, // (existingOptions: object = {}) => object
+  isHidden: PropTypes.bool, // `true` to hide from Edit visualization dialog
+  isDeprecated: PropTypes.bool,
+  Renderer: PropTypes.func.isRequired,
```

Comment:
Maybe `RenderComponent` to make it more explicit? (and same for editor: `EditorComponent`)?

Reply:
1. I called `Renderer` in this way to keep semantic convention: a visualization Render**er**, a visualization Edit**or**;
2. Adding `Component` suffix sounds like a tautology ("information IT-technologies", real-life example from times I was student. yeah...). But of course I'll appreciate any ideas for better names for that components.